intermine/intermine
# Export datapackage

## Details

Exports a frictionless [tabular data package](https://specs.frictionlessdata.io/tabular-data-package/#example) with a [tabular data resource](https://specs.frictionlessdata.io/tabular-data-resource/#examples).

## Testing

1. Choose any template or build a query.
2. While exporting check the **export frictionless datapackage** checkbox.
3. Click on the **Download file** button.

Clone the im-tables repo locally and follow these steps to test the feature -

[This PR](https://github.com/intermine/im-tables/pull/237) in im-tables repo contains the updated code which should be used with this PR to test the feature.

1. Download ngrok from [here](https://ngrok.com/download).
2. Run im-tables locally using a. `npm install` and b. `npm run start`.
3. Run ngrok on another terminal using `ngrok http 9000`.
4. Change the URL of im-tables location in line 72 of global.web.properties file. 

## Checklist

Before your pull request can be approved, be sure to check all boxes:

- [ ] Passing unit test for new or updated code (if applicable)
- [x] Passes all tests – according to Travis
- [x] Single purpose
- [x] Well commented code
- [x] Checkstyle
